### PR TITLE
chore: bump spin version to 1.5.0

### PR DESCRIPTION
**Pull request by jprendes**
Bump spin `1.4.1` -> `1.5.0`.

### CODE REVIEW

**Comment by jprendes** [comment]
@cpuguy83 @devigned PTAL

**Review by cpuguy83 — APPROVED** [review]

**Review by Mossaka — APPROVED** [review]
lgtm! But it looks like the CI failed

**Comment by jprendes** [comment]
IIUC, there's a problem building a dependency introduced by the new LLM feature of spin.

**Comment by jprendes** [comment]
Ok, I digged a bit into it.
What's failing is building `llama-cpp` in `aarch64` `musl`.
It builds fine with `gnu` ABI adding the `-Ctarget-feature=+fp16` (what upstream spin does).
But it fails with `musl` ABI.
There are a few diferences, including the `gcc` version that `cross` uses for those targets.

The main issue is that `llama-cpp` is using the  `vld1q_u8_x4` and `vld1q_s8_x4` functions, which should be SIMD intrinsics exposed by the compiler.

According to this issue https://github.com/xmrig/xmrig/pull/2882, that intrinsic is not present in `gcc` <= 11.

I am wondering what the right fix is, since `spin` doesn't provide a feature flag to disable llm.

File: rust-toolchain.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[toolchain]
```
Comment (Mossaka) [inline rust-toolchain.toml:1]: Will this file be used?

**Review by Mossaka — APPROVED** [review]

File: rust-toolchain.toml (hunk 1 above)
Comment (jprendes) [reply rust-toolchain.toml:1]: I do hope so. Is your concern that the file is not in the same directory as the Cargo.toml files?
The [doc](https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file) is a bit ambiguous, but I do hope it gets used!

**Review by Mossaka — APPROVED** [review]
lgtm! 

minor comment: add a readme on `polyfill` folder to describe what this does and when shall we remove it. 

This can be done in a future PR.

File: rust-toolchain.toml (hunk 1 above)
Comment (Mossaka) [reply rust-toolchain.toml:1]: > Is your concern that the file is not in the same directory as the Cargo.toml files?

precisely. I am not sure either.

**Comment by Mossaka** [comment]
Need a rebase

File: rust-toolchain.toml (hunk 1 above)
Comment (jprendes) [reply rust-toolchain.toml:1]: I guess the answer is yes, otherwise CI would have failed :-)

**Comment by jprendes** [comment]
> minor comment: add a readme on polyfill folder to describe what this does and when shall we remove it.

If there's an update to the compiler in cross-rs, the polyfill will become a no-op as it does a feature test before acting.
I left a link to the upstream issue in spin.
There are many different solutions, off the top of my head I can think of:
* a feature gate in spin
* a compiler update in cross
* replacing cross with something like xx, similar to what we do in runwasi (although cross is really nice, but would also allow us to build with systemd and seccomp support here)

**Review by Mossaka — APPROVED** [review]

**Comment by jprendes** [comment]
sorry, changed laptops and hadn't set up git correctly!
it's signed now :-)

**Pull request merged**